File: solutions/r/perfect-numbers/1/perfect-numbers.R
number_type <- function(n){
	factors <- function (n) {
		if (n == 1) {return (c())}
		for (d in 1:(n %/% 2)) {
			if (d == 1) {
				f <- c(1) 
			} else if  (n %% d == 0)
			{
				f <- c(f, d)
			}			
		}
		return (f)
	}
	sf <- sum(factors (n))
	if (sf == n) 
	{ 
		return ("perfect") 
	} else if (sf > n) 
	{
		return ("abundant")
	} else 
	{
		return ("deficient")
	}
}

File: solutions/python/perfect-numbers/1/perfect_numbers.py
def classify(number):
    if number < 1:
        raise ValueError("Value Error")
    aliquot_sum = sum(factors(number))
    if aliquot_sum == number:
        return("perfect")
    elif aliquot_sum > number:
        return("abundant")
    else: 
        return("deficient")

def factors(number):
    return([i for i in range(1,int(number / 2)+1) if number % i == 0])

